Functions/Example/Tests/FIRFunctionsTests.m

```diff
@@ -34,7 +34,7 @@ - (void)tearDown {
 
 - (void)testURLWithName {
   id app = [[FUNFakeApp alloc] initWithProjectID:@"my-project"];
-  FIRFunctions *functions = [FIRFunctions functionsForApp:app region:@"my-region"];
+  FIRFunctions *functions = [[FIRFunctions alloc] initWithApp:app region:@"my-region" auth:nil];
```

Comment:
Can this keep the constructor it had? It's nice to have a test of the constructor, and it doesn't seem like we need to pass auth here.

Reply:
By removing the app from the designated initializer and deleting `FUNFakeApp`, this class factory method no longer works for this test.

Since the test is for `URLWithName:`, this seems reasonable to me. If we wish to test the behaviour of the convenience initializers, I would recommend separate tests using OCMock to modify the default app or use a partial mock of an app that verifies the given fields are accessed as expected instead of having a custom fake app instance.

Reply:
Yes, it makes sense for this to change now that `FUNFakeApp` is going away.